www/public/index.html

```diff
@@ -41,8 +55,241 @@ <h1>Work in Progress</h1>
     <li><a href="https://youtu.be/ZnYa99QoznE?t=4790">Roc at the Berlin FP Meetup</a> - September 1, 2020 (overall
       vision for the language)</li>
   </ul>
-  <p>Roc also compiles to WebAssembly and you can try it out on the web <a href="repl/">here</a> (remembering it's not ready yet!)</p>
-  <p>You can check out the <a href="https://github.com/roc-lang/roc">source code repository</a> if you'd like to play around with it in its current state.</p>
+
+  <h2>A <em>Fast</em> Language</h2>
+
+  <h3>Goal</h3>
+
+  <p>The goal is that Roc runs so fast that in practice, you wouldn't expect a significant
+    performance boost from switching to a more popular language unless that language runs
+    at least as fast as C.</p>
+
+  <p>When benchmarking Roc code against similarly-optimized <a href="https://go.dev">Go</a>,
+    <a href="https://www.swift.com/">Swift</a>, <a href="https://www.java.com">Java</a>,
+    <a href="https://learn.microsoft.com/en-us/dotnet/csharp">C#</a>, and
+    <a href="https://www.ecma-international.org/publications-and-standards/standards/ecma-262">JavaScript</a>,
+    we generally aim for Roc to outperform all of those languages. Outperforming systems
+    languages like Rust, Zig, C, D, and C++ is a non-goal, as is outperforming research languages
+    that see little or no use in industry. (Presumably there will always be certain specific
+    benchmarks where some popular slower-than-C languages outperform Roc, but the goal is to
+    usually be at the front of that pack.)</p>
+
+  <h4>Current progress</h4>
+
+  <p>Progress towards this performance goal is already quite far along.</p>
+
+  <p>Since Roc already uses unboxed data structures that get monomorphized into LLVM—just like Rust,
+    Zig, and C++ do—in many cases Roc code already compiles to the same machine instructions that the
+    equivalent code written in one of these systems languages would. Something we do regularly is
+    to compare the LLVM instructions generated by Roc's compiler and by these systems languages'
+    compilers, to check whether we're generating equivalent instructions.</p>
+
+  <p>That said, there are also cases where Roc has strictly more runtime overhead than languages
+    like C, C++, Zig, and Rust do. The most significant is automatic memory management, which Roc
+    implements using automatic reference counting. Static reference count optimizations like
+    elision and reuse (thanks to Morphic and
+    <a href="https://www.microsoft.com/en-us/research/publication/perceus-garbage-free-reference-counting-with-reuse/">Perceus</a>)
+    improve things, but significant runtime overhead remains.</p>
```

Comment:
nit, pedantic: significant was used in the last sentence, using it again here reads a bit strange to me